sphinx_immaterial/apidoc/python/apigen.py

```diff
@@ -0,0 +1,1908 @@
+            member_name=member_name,
+            member_value=member_value,
+            is_attr=is_attr,
+        )
+        if member_documenter is None:
+            continue
+        full_name = f"{documenter.fullname}.{member_name}"
+        entry = _MemberDocumenterEntry(
+            cast(sphinx.ext.autodoc.Documenter, member_documenter),
+            is_attr,
+            parent_canonical_full_name=canonical_full_name,
+            name=member_name,
+            full_name=full_name,
+        )
+        for transformed_entry in _transform_member(documenter, entry):
+            yield from _get_member_overloads(transformed_entry)
+
+
+def _get_documenter_members(
+    documenter: sphinx.ext.autodoc.Documenter,
+    canonical_full_name: str,
+) -> Iterator[_MemberDocumenterEntry]:
+    """Returns the sequence of members to document, including inherited members.
+
+    :param documenter: Parent documenter for which to find members.
+    :returns: Iterator over members to document.
+    """
+    seen_members: Set[str] = set()
+
+    def _get_unseen_members(
+        members: Iterator[_MemberDocumenterEntry], is_inherited: bool
+    ) -> Iterator[_MemberDocumenterEntry]:
+        for member in members:
+            overload_name = member.toc_title
+            if overload_name in seen_members:
+                continue
+            seen_members.add(overload_name)
+            yield member._replace(is_inherited=is_inherited)
+
+    yield from _get_unseen_members(
+        _get_documenter_direct_members(
+            documenter, canonical_full_name=canonical_full_name
+        ),
+        is_inherited=False,
+    )
+
+    if documenter.objtype != "class":
+        return
+
+    for cls in inspect.getmro(documenter.object):
+        if cls is documenter.object:
+            continue
+        if cls.__module__ in ("builtins", "pybind11_builtins"):
+            continue
+        class_name = f"{cls.__module__}::{cls.__qualname__}"
+        parent_canonical_full_name = f"{cls.__module__}.{cls.__qualname__}"
+        try:
+            superclass_documenter = _create_documenter(
+                env=documenter.env,
+                documenter_cls=sphinx.ext.autodoc.ClassDocumenter,
+                name=class_name,
+                tab_width=documenter.directive.state.document.settings.tab_width,
+            )
+            yield from _get_unseen_members(
+                _get_documenter_direct_members(
+                    superclass_documenter,
+                    canonical_full_name=parent_canonical_full_name,
+                ),
+                is_inherited=True,
+            )
+        except Exception as e:  # pylint: disable=broad-except
+            logger.warning(
+                "Cannot obtain documenter for base class %r of %r: %r",
+                cls,
+                documenter.fullname,
+                e,
+            )
+
+
+_ENTITY_PAGE_INITIAL_COMMENT = (
+    "..\n  DO NOT EDIT. GENERATED by sphinx-immaterial python_apigen.\n"
```

Comment:
This seems a bit outdated since the domain-centric refactor.
```suggestion
    "..\n  DO NOT EDIT. GENERATED by sphinx_immaterial.apidoc.python.apigen.\n"
```